core/cat/infrastructure/package.py

```diff
@@ -19,13 +20,26 @@ def __init__(self, path):
         self.path = path
 
     def unpackage(self, to):
+
+        # list of folder contents before extracting
+        ls_before = os.listdir(to)
+
+        # extract
         if self.extension == 'zip':
             with zipfile.ZipFile(self.path, 'r') as zip_ref:
                 zip_ref.extractall(to)
         elif self.extension == 'tar':
             with tarfile.open(self.path, 'r') as tar_ref:
                 tar_ref.extractall(to)
 
+        # list of folder contents after extracting
+        ls_after = os.listdir(to)
+
+        # return extracted contents paths
+        # TODO: does not handle overwrites, should extract in a /temp new folder and then copy to destination
+        ls = set(ls_after) - set(ls_before)
```

Comment:
Everything seems fine to me, I just don't understand this part

Reply:
`ls_before` = what was in the destination folder before extraction
`ls_after` = what was after
`ls` = list of contents extracted

Can be done better, it works but does not take into account if some contents are overwritten

Reply:
`set` in python is a native data structure representing a set of unique elements
`setA - setB` returns elements in setA that are not present in setB